src/types/Currency.sol

```diff
@@ -31,12 +31,12 @@ library CurrencyLibrary {
     using CustomRevert for bytes4;
 
     /// @notice Thrown when a native transfer fails
-    /// @param revertReason bubbled up revert reason
-    error NativeTransferFailed(bytes revertReason);
+    /// @param reason bubbled up revert reason
+    error Wrap__NativeTransferFailed(address recipient, bytes reason);
 
     /// @notice Thrown when an ERC20 transfer fails
-    /// @param revertReason bubbled up revert reason
-    error ERC20TransferFailed(bytes revertReason);
+    /// @param reason bubbled up revert reason
+    error Wrap__ERC20TransferFailed(address token, address recipient, bytes reason);
```

Comment:
is this right? looks like its only passing in 1 address not two

Reply:
ugh, thanks for catching it!